keywords/TestServeriOS.py

```diff
@@ -0,0 +1,323 @@
+
+    def remove(self):
+        """
+        Remove the iOS app from the simulator
+        """
+        bundle_id = "com.couchbase.CBLTestServer-iOS"
+        log_info("Removing CBLTestServer")
+
+        self.stop()
+
+        # Stop the simulator
+        log_info("device_id: {}".format(self.device_id))
+        output = subprocess.check_output([
+            "killall", "Simulator"
+        ])
+
+        # Erase the simulator
+        output = subprocess.check_output([
+            "xcrun", "simctl", "erase", self.device_id
+        ])
+
+        if bundle_id in output:
+            raise LiteServError("{} is still present after uninstall".format(bundle_id))
+
+    def start(self, logfile_name):
+        """
+        1. Starts a LiteServ with logging to provided logfile file object.
+           The running LiteServ process will be stored in the self.process property.
+        2. The method will poll on the endpoint to make sure LiteServ is available.
+        3. The expected version will be compared with the version reported by http://<host>:<port>
+        4. Return the url of the running LiteServ
+        """
+
+        # self.device = "iPhone-7-Plus"
+        self.logfile_name = logfile_name
+
+        # package_name = "CBLTestServer-iOS.app"
+        # app_dir = "CBLTestServer-iOS"
+
+        self.app_path = "{}/{}/{}".format(BINARY_DIR, self.app_dir, self.app_name)
+        # self.app_path = "/Users/sridevi.saragadam/Library/Developer/Xcode/DerivedData/CBLTestServer-iOS-ayypbvnmphhpihebbhrjusdqhzmk/Build/Products/Release-EE-iphonesimulator/CBLTestServer-iOS.app"
+        # Without --exit, ios-sim blocks
+        # With --exit, --log has no effect
+        # subprocess.Popen didn't launch the app
+        output = subprocess.check_output([
+            "ios-sim", "--devicetypeid", self.device, "launch", self.app_path, "--exit"
+        ])
+
+        log_info(output)
+        self._wait_until_reachable(port=self.port)
+        # time.sleep(10)
+        self._verify_running()
+        # return "http://{}:{}".format(self.host, self.port)
+
+    def start_device(self, logfile_name):
+        """
+        1. Starts a LiteServ with logging to provided logfile file object.
+           The running LiteServ process will be stored in the self.process property.
+        2. The method will poll on the endpoint to make sure LiteServ is available.
+        3. The expected version will be compared with the version reported by http://<host>:<port>
+        4. Return the url of the running LiteServ
+        """
+
+        self.logfile_name = logfile_name
+        self.app_path = "{}/{}/{}".format(BINARY_DIR, self.app_dir, self.app_name)
+
+        output = subprocess.check_output([
+            "ios-deploy", "--justlaunch", "--bundle", self.app_path
+        ])
+        log_info(output)
+
+        self._wait_until_reachable(port=self.port)
+        self._verify_running()
+
+        # return "http://{}:{}".format(self.host, self.port)
```

Comment:
remove the commented code, if not needed.

Reply:
done